### ISSUE TITLE
De-couple job enqueueing

### ISSUE THREAD

**Issue by webignition** [webignition/worker.simplytestable.com#173]
**Overview**
Some actions are followed by the enqueueing of jobs to allow other actions to be performed asynchronously.

It is assumed that the actions being performed prior to enqueueuing are synchronous. This is soon not to be the case with task preparation and task performance.

**Approximate solution**
A solution involving the event dispatcher is needed.

For all cases where a resque job is enqueued, we need instead to dispatch an event to say that something has happened and then have an event listener (or subscriber) that then performs the enqueuement.

This allows for the enqueuement to  be deferred until an asynchronous operation has completed.

**Example**
`PrepareCommand` currently contains:

```php
$taskId = $this->task->getId();
$this->taskPreparer->prepare($this->task);

$this->resqueQueueService->enqueue(new TaskPerformJob(['id' => $taskId]));
```

The above code assumes `$this->taskPreparer->prepare($this->task)` to be synchronous. 

It will soon be the case that task preparation will utilise asynchronous http resource retrieval making the task preparer operation asynchronous. 

We will then need to remove the enqueuement from `PrepareCommand`:

```php
$taskId = $this->task->getId();
$this->taskPreparer->prepare($this->task);
```

It will be up to the task preparer to dispatch a similar event once asynchronous preparation is complete:

```php
$this->eventDispatcher->dispatch('task.prepared', new TaskPreparedEvent());
```
